scene/3d/skeleton_modifier_3d.cpp

```diff
@@ -0,0 +1,158 @@
+/* SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                 */
+/**************************************************************************/
+
+#include "skeleton_modifier_3d.h"
+
+void SkeletonModifier3D::_validate_property(PropertyInfo &p_property) const {
+	//
+}
+
+PackedStringArray SkeletonModifier3D::get_configuration_warnings() const {
+	PackedStringArray warnings = Node3D::get_configuration_warnings();
+	if (skeleton_id.is_null()) {
+		warnings.push_back(RTR("Skeleton3D node not set! SkeletonModifier3D must be child of Skeleton3D or set a path to an external skeleton."));
+	}
+	return warnings;
+}
+
+void SkeletonModifier3D::rebind() {
+	_update_skeleton();
+	_rebind();
+}
+
+void SkeletonModifier3D::_rebind() {
+	//
+}
+
+/* Skeleton3D */
+
+Skeleton3D *SkeletonModifier3D::get_skeleton() const {
+	return Object::cast_to<Skeleton3D>(ObjectDB::get_instance(skeleton_id));
+}
+
+void SkeletonModifier3D::_update_skeleton_path() {
+	skeleton_id = ObjectID();
+
+	// Make sure parent is a Skeleton3D.
+	Skeleton3D *sk = Object::cast_to<Skeleton3D>(get_parent());
+	if (sk) {
+		skeleton_id = sk->get_instance_id();
+	}
+}
+
+void SkeletonModifier3D::_update_skeleton() {
+	if (!is_inside_tree()) {
+		return;
+	}
+	Skeleton3D *old_sk = get_skeleton();
+	_update_skeleton_path();
+	Skeleton3D *new_sk = get_skeleton();
+	if (old_sk != new_sk) {
+		_skeleton_changed(old_sk, new_sk);
+	}
+	update_configuration_warnings();
+}
+
+void SkeletonModifier3D::_skeleton_changed(Skeleton3D *p_old, Skeleton3D *p_new) {
+	//
+}
+
+/* Process */
+
+void SkeletonModifier3D::set_active(bool p_active) {
+	if (active == p_active) {
+		return;
+	}
+	active = p_active;
+	_set_active(active);
+}
+
+bool SkeletonModifier3D::is_active() const {
+	return active;
+}
+
+void SkeletonModifier3D::_set_active(bool p_active) {
+	//
+}
+
+void SkeletonModifier3D::set_influence(real_t p_influence) {
+	influence = p_influence;
+}
+
+real_t SkeletonModifier3D::get_influence() const {
+	return influence;
+}
+
+void SkeletonModifier3D::process_modification() {
+	if (!active) {
+		return;
+	}
+	GDVIRTUAL_CALL(_process_modification);
+	_process_modification();
+	emit_signal(SNAME("modification_processed"));
+}
+
+void SkeletonModifier3D::_process_modification() {
+	//
+}
+
+void SkeletonModifier3D::_notification(int p_what) {
+	switch (p_what) {
+		case NOTIFICATION_ENTER_TREE:
+		case NOTIFICATION_PARENTED: {
+			rebind();
+		} break;
+	}
+}
+
+void SkeletonModifier3D::_bind_methods() {
+	ClassDB::bind_method(D_METHOD("set_active", "active"), &SkeletonModifier3D::set_active);
+	ClassDB::bind_method(D_METHOD("is_active"), &SkeletonModifier3D::is_active);
+
+	ClassDB::bind_method(D_METHOD("set_influence", "influence"), &SkeletonModifier3D::set_influence);
+	ClassDB::bind_method(D_METHOD("get_influence"), &SkeletonModifier3D::get_influence);
+
+	ClassDB::bind_method(D_METHOD("_process_modification"), &SkeletonModifier3D::_process_modification);
```

Comment:
```suggestion
	ClassDB::bind_method(D_METHOD("process_modification"), &SkeletonModifier3D::process_modification);
```

Reply:
Without this change, GDScript will not permit me to override `_process_modification`. I suspect this was a mistake since users who wish to invoke the method should call `process_modification()`

Reply:
I remember agreeing with @reduz that this should not have been allowed in the first place. Remove it once and add a method for expansion later.

Reply:
I do not understand here what should not be allowed.
Anyway there is just a bug here:
```
ClassDB::bind_method(D_METHOD("_process_modification"), &SkeletonModifier3D::_process_modification);
GDVIRTUAL_BIND(_process_modification);
```
is not allowed due to defining `_process_modification`. This causes a warning in GDScript which can be treated as error, so that's why it should be fixed.

Reply:
There is zero performance cost to permitting GDVIRTUAL from the base class, so that reasoning is incorrect.

Anyway it doesn't matter. I'll patch it back in my build for now: I can't test this PR without virtual, since there is no other way to implement custom modifiers.